v4/newrelic/transaction.go

```diff
@@ -220,11 +235,40 @@ func (txn *Transaction) InsertDistributedTraceHeaders(hdrs http.Header) {}
 // AcceptDistributedTraceHeaders first looks for the presence of W3C trace
 // context headers.  Only when those are not found will it look for the New
 // Relic distributed tracing header.
-func (txn *Transaction) AcceptDistributedTraceHeaders(t TransportType, hdrs http.Header) {}
+func (txn *Transaction) AcceptDistributedTraceHeaders(t TransportType, hdrs http.Header) {
+	if txn == nil {
+		return
+	}
+
+	txn.thread.currentSpan.Lock()
+	defer txn.thread.currentSpan.Unlock()
```

Comment:
Similarly, this line will panic if the transaction is a `&Transaction{}`.

Reply:
Fixed, and added a test.